membership_prorate_variable_period/models/product_template.py

```diff
@@ -0,0 +1,28 @@
+# Copyright 2016 Tecnativa - Antonio Espinosa
+# Copyright 2019 Tecnativa - David Vidal
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+import math
+from odoo import _, api, exceptions, models
+from odoo.tools import date_utils
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    @api.multi
+    def _get_next_date(self, date, qty=1):
+        next_date = super(ProductTemplate, self)._get_next_date(date)
+        if self.membership_interval_unit == 'days':
+            raise exceptions.Warning(
+                _("It's not possible to prorate daily periods."))
+        qty = int(math.ceil(qty)) * self.membership_interval_qty
```

Comment:
math.ceil return an integer, cast it is redundant.

Reply:
Sure :) changed